src/evaluator/functions.rs

```diff
@@ -1089,6 +1089,69 @@ pub fn to_millis<'a>(
     }
 }
 
+pub fn fn_zip<'a>(
+    context: FunctionContext<'a, '_>,
+    args: &[&'a Value<'a>],
+) -> Result<&'a Value<'a>> {
+    if args
+        .iter()
+        .any(|arg| arg.is_null() || matches!(arg, Value::Undefined))
+    {
+        return Ok(context.arena.alloc(Value::Array(
+            bumpalo::vec![in context.arena;],
+            ArrayFlags::empty(),
+        )));
+    }
+
+    // Create a vector of arrays
+    let arrays: Vec<&bumpalo::collections::Vec<'a, &'a Value<'a>>> = args
+        .iter()
+        .filter_map(|arg| match arg {
+            Value::Array(arr, _) => Some(arr),
+            _ => None,
+        })
+        .collect();
+
+    if arrays.is_empty() {
+        let mut result: bumpalo::collections::Vec<&Value<'a>> = bumpalo::vec![in context.arena;];
+        let mut outer_array: bumpalo::collections::Vec<&Value<'a>> =
+            bumpalo::vec![in context.arena;];
+        for arg in args {
+            result.push(*arg);
+        }
+        outer_array.push(
+            context
+                .arena
+                .alloc(Value::Array(result, ArrayFlags::empty())),
+        );
+        return Ok(context
+            .arena
+            .alloc(Value::Array(outer_array, ArrayFlags::empty())));
+    }
+
+    let min_length = arrays.iter().map(|arr| arr.len()).min().unwrap_or(0);
```

Comment:
I wonder if it is possible to use the standard libraries `Iterator::zip` to implement the remaining of this function. We might also be able to collect the results into a `bumpalo::collections::Vec`.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip
https://docs.rs/bumpalo/latest/bumpalo/collections/trait.FromIteratorIn.html#tymethod.from_iter_in